### DISCUSSION TITLE
[feature request] resource route discovery

### DISCUSSION THREAD

**Discussion by menthol** [spatie/laravel-route-discovery#14] [Ideas]
This package work great but required to set attributes to almost all my action because I use a lot of nested resource.

I suggest that we can add a resource route discovery.

```php
<?php

Discover::resources()->in($directory);
```
```bash
$ artisan route:list --except-vendor
GET|HEAD   users ............................................. users.index > UserController@index
GET|HEAD   users/dummy ....................................... users.dummy > UserController@dummy
POST       users/{user} ...................................... users.store > UserController@store
GET|HEAD   users/{user}/friends ............... users.friends.index > User\FriendController@index
POST       users/{user}/friends/request ... users.friends.request > User\FriendController@request
GET|HEAD   users/{user}/friends/{friend} ........ users.friends.show > User\FriendController@show
```

**Comment by elnurvl** [comment]
Why would `UserController@store` need a `/{user}` appended to the route? To store a user, you would do a `POST` request to `users/` endpoint with the parameters in the request body.

But yes, I would really like to see this discussion to get an attention from the package maintainers(@freekmurze). This would be huge for a resource-based REST API development workflow.

**Comment by ricventu** [comment]
Hi, 
you can do this, by creating subdirectory with the name of the resource

[https://github.com/spatie/laravel-route-discovery/pull/36](url)
